modules/migrations/gitlab_test.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2019 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"code.gitea.io/gitea/modules/migrations/base"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGitlabDownloadRepo(t *testing.T) {
+	// Skip tests if Gitlab token is not found
+	gitlabPersonalAccessToken := os.Getenv("GITLAB_READ_TOKEN")
+	if gitlabPersonalAccessToken == "" {
+		t.Skip("skipped test because GITLAB_READ_TOKEN was not in the environment")
+	}
+
```

Comment:
```suggestion
resp, err := http.Get("https://gitlab.com/gitea/test_repo")
	if err != nil || resp.StatusCode/100 != 2 {
		t.Skipf("Can't access test repo, skipping %s", t.Name())
	}
```
Untested but something like this to skip the test if the repo isn't accessible (would need import "net/http" too)

Reply:
Good thinking. That seemed to work in my rudimentary testing.